library/multichoicerecyclerview/src/main/java/com/davidecirillo/multichoicerecyclerview/MultiChoiceAdapter.java

```diff
@@ -80,8 +83,14 @@ public void selectAll() {
      * @return true if has been selected false otherwise
      */
     public boolean select(int position) {
-        if (mIsInMultiChoiceMode || mIsInSingleClickMode && mItemList.containsKey(position)) {
-            perform(Action.SELECT, position, true, true);
+        if (mItemList.containsKey(position)) {
+            if (mIsInSingleClickMode || mIsInMultiChoiceMode) {
+                perform(Action.SELECT, position, true, true);
+                return true;
+            }
+        } else {
+            mItemList.put(position, State.INACTIVE);
```

Comment:
Why is this check needed in here (and in `processUpdate()` and `processClick`)? There won't be a problem, when the list is populated before those methods are called, right?

So I would suggest, just doing a nllptr-check an return (doing nothing) if the given position is not in the list.

When coming from an orientation-change, the first thing called is `onRestoreInstanceState()` which restores the whole list so everything should be fine.

Have you tried is this way?